### PR TITLE
WIP: Fix useMediaQuery to support Safari

### PR DESCRIPTION
**Pull request by dpschen**
I had problems using useMediaQuery on an iPhone 7. With the changes I could get it working.

### CODE REVIEW

**Comment by antfu** [comment]
Thanks! Please un-draft it when you finish.

File: packages/core/useMediaQuery/index.ts [hunk 1]
```diff
@@ -15,10 +36,10 @@ export function useMediaQuery(query: string) {
     matches.value = event.matches
   }
 
-  mediaQuery.addEventListener('change', handler)
+  matchMediaListener(mediaQuery, handler)
 
   tryOnUnmounted(() => {
-    mediaQuery.addEventListener('change', handler)
+    matchMediaListener(mediaQuery, handler)
```
Comment (dpschen) [inline packages/core/useMediaQuery/index.ts:42]: @antfu Maybe I misunderstand here something. Shouldn't the event be removed in tryOnUnmounted?

Comment (antfu) [reply packages/core/useMediaQuery/index.ts:42]: Yeah, was a typo here, thanks

**Comment by anteriovieira** [comment]
Hi @dpschen , thank you very much for your effort in solving this problem.

We are closing this pull request, it was solved in #164

**Comment by antfu** [comment]
Close in favor of #164. Thanks

**Comment by dpschen** [comment]
Much better implemention! @anteriovieira  :)
Sry I couldn't finish this earlier @antfu . There was a lot on my place recently

**Pull request closed without merging**